dev_docs/POLICIES.md

```diff
@@ -0,0 +1,95 @@
+- Warnings of future compatibility changes, which are not "deprecate and remove", may be introduced in a `major` or `minor` release. Future compatibility warnings must be included in the appropriate change log.
+- A `patch` release may not introduce deprecations or future incompatibility change warnings.
+
+Experimental features may be modified in a `major` or `minor` release without prior warning. However, where possible, advance warning of changes will be provided.
+
+## Warning period
+
+Breaking API changes will provide notice of at least two `minor` releases or six months, whichever is greater, before effecting the change.
+
+## Implementing Deprecation
+
+### Python deprecations
+
+Upon deprecation of Python functionality:
+
+- a `DeprecationWarning` must be issued by the deprecated functionality and a unit test provided to confirm warning issuance.
+- affected lifecyle tags must be set to "deprecated", typically in the docstring and other user-visible documentation.
+- docstrings must clearly indicate the scope of deprecation (e.g, function, a particular set of parameters, etc) and must include actionable guidance on migration.
+- HISTORY.md must clearly indicate the deprecation and any guidance on migration.
+
+For other breaking changes which are not "deprecate and remove":
+
+- a `FutureWarning` must be issued by the deprecated functionality and a unit test provided to confirm warning issuance.
+- docstrings must clearly indicate the scope of the breaking change (e.g, function, a particular set of parameters, etc) and must include actionable guidance on migration
+- HISTORY.md must clearly indicate the breaking change and any guidance on migration.
+
+Where entire classes, methods or functions are deprecated:
+
+- utilize `typing_extensions.deprecated` decorator to provide both static and runtime warnings.
+- update unit tests to ensure that the warning is issued (e.g., utilize `pytest.deprecated_call()`).
+- ensure that `stacklevel` is set so that the warning comes from the invoking (caller's) code.
+
+### R deprecations
+
+Upon deprecation of R functionality, methods, or functions:
+
+- a deprecation warning generated by `lifecycle::deprecate_warn()` must be issued by the deprecated functionality, and a unit test `lifecycle::expect_deprecated()` provided to confirm warning issuance.
+- if any parameters are being deprecated, they must be given a default value of `lifecycle::deprecated()`
+- documentation must clearly indicate the scope of deprecation (e.g, function, a particular set of parameters, etc) and must include actionable guidance on migration.
+- `NEWS.md` must clearly indicate the deprecation and any guidance on migration.
+
+R functionality additionaly goes through a defunction process begining two `minor` releases after the deprecation
```

Comment:
```suggestion
R functionality additionally goes through a defunction process beginning two `minor` releases after the deprecation.
```

I think we should spell out what becoming defunct means in practice (will the function throw an error and stop working, or?).

For alignment what about the "or three months, whichever is greater"? If two minor releases happen in less than three months, it would be defunct before the three-month mark.

Maybe something like:

> Deprecated R functionality will become defunct
after two `minor` releases or 3 months, whichever is greater.